resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -162,6 +187,8 @@ public DnsNameResolver(
         this.optResourceEnabled = optResourceEnabled;
         this.hostsFileEntriesResolver = checkNotNull(hostsFileEntriesResolver, "hostsFileEntriesResolver");
         this.resolveCache = resolveCache;
+        this.searchDomains = checkNotNull(searchDomains, "searchDomains");
```

Comment:
we will need to call `searchDomains.clone()` as we can not be sure the caller will not modify the array later. Yay for java that not has `const` :(